great_expectations/rule_based_profiler/toolkit.py

```diff
@@ -2,18 +2,80 @@
 from typing import List, Optional, Union
 
 import great_expectations.exceptions as ge_exceptions
+from great_expectations.checkpoint.util import batch_request_contains_batch_data
 from great_expectations.data_context.store import ProfilerStore
 from great_expectations.data_context.types.base import DataContextConfigDefaults
+from great_expectations.data_context.types.refs import GeCloudIdAwareRef
 from great_expectations.data_context.types.resource_identifiers import (
     ConfigurationIdentifier,
     GeCloudIdentifier,
 )
 from great_expectations.data_context.util import instantiate_class_from_config
 from great_expectations.rule_based_profiler import RuleBasedProfiler
-from great_expectations.rule_based_profiler.config import RuleBasedProfilerConfig
+from great_expectations.rule_based_profiler.config.base import RuleBasedProfilerConfig
 from great_expectations.util import filter_properties_dict
 
 
+def add_profiler(
+    config: RuleBasedProfilerConfig,
+    data_context: "DataContext",  # noqa: F821
+    ge_cloud_id: Optional[str] = None,
+) -> RuleBasedProfiler:
+    if not _check_validity_of_batch_requests_in_config(config):
+        raise ge_exceptions.InvalidConfigError(
+            f'batch_data found in batch_request cannot be saved to ProfilerStore "{data_context.profiler_store_name}"'
+        )
+
+    profiler_config: dict = config.to_json_dict()
+
+    new_profiler = instantiate_class_from_config(
```

Comment:
@cdkini Type hint, please -- for aesthetics purposes.  Thanks! :)

Reply:
I agree with it's placement here because `instantiate_class_from_config` returns any business logic object.